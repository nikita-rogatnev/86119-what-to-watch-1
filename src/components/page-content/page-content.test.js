import React from 'react';
import renderer from 'react-test-renderer';

import PageContent from './page-content';

const mock = {
  films: [
    {
      id: 1,
      name: `Fantastic Beasts: The Crimes of Grindelwald`,
      link: `movie-page.html`
    },
    {
      id: 2,
      name: `Bohemian Rhapsody`,
      link: `movie-page.html`
    },
    {
      id: 3,
      name: `Macbeth`,
      link: `movie-page.html`
    },
  ],
  genres: [
    {
      "title": `All genres`,
      "isActive": true
    },
    {
      "title": `Comedies`,
      "isActive": false
    },
    {
      "title": `Crime`,
      "isActive": false
    },
  ]
};

it(`PageContent renders correctly`, () => {
  const {films, genres} = mock;

  const tree = renderer.create(<PageContent films={films} genres={genres}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
