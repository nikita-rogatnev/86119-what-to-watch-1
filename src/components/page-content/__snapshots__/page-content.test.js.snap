// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContent renders correctly 1`] = `
<div
  className="page-content"
>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <ul
      className="catalog__genres-list"
    >
      <li
        className="catalog__genres-item catalog__genres-item--active"
      >
        <a
          className="catalog__genres-link"
          href="#"
        >
          All genres
        </a>
      </li>
      <li
        className="catalog__genres-item "
      >
        <a
          className="catalog__genres-link"
          href="#"
        >
          Comedies
        </a>
      </li>
      <li
        className="catalog__genres-item "
      >
        <a
          className="catalog__genres-link"
          href="#"
        >
          Crime
        </a>
      </li>
    </ul>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
      >
        <button
          className="small-movie-card__play-btn"
          type="button"
        >
          Play
        </button>
        <div
          className="small-movie-card__image"
        >
          <img
            alt="Fantastic Beasts: The Crimes of Grindelwald"
            height="175"
            src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
            onClick={[Function]}
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <button
          className="small-movie-card__play-btn"
          type="button"
        >
          Play
        </button>
        <div
          className="small-movie-card__image"
        >
          <img
            alt="Bohemian Rhapsody"
            height="175"
            src="img/bohemian-rhapsody.jpg"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
            onClick={[Function]}
          >
            Bohemian Rhapsody
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <button
          className="small-movie-card__play-btn"
          type="button"
        >
          Play
        </button>
        <div
          className="small-movie-card__image"
        >
          <img
            alt="Macbeth"
            height="175"
            src="img/macbeth.jpg"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
            onClick={[Function]}
          >
            Macbeth
          </a>
        </h3>
      </article>
    </div>
    <div
      className="catalog__more"
    >
      <button
        className="catalog__button"
        type="button"
      >
        Show more
      </button>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
